include/iceflow/iceflow.hpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2024 The IceFlow Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ICEFLOW_HPP
+#define ICEFLOW_HPP
+
+#include "constants.hpp"
+#include "data.hpp"
+#include "logger.hpp"
+#include "ndn-svs/security-options.hpp"
+#include "ringbuffer.hpp"
+#include <ndn-svs/svspubsub.hpp>
+
+#include <iostream>
+#include <thread>
+
+namespace iceflow {
+
+/**
+ * Central building block for IceFlow-based consumers and producers.
+ *
+ * Will act as a consumer if a `subTopic` is defined and act as a consumer if
+ * a `pubTopic` is defined.
+ * If both topics are passed to the constructor, the instantiated IceFlow
+ * object will act both as a producer _and_ a consumer.
+ */
+class IceFlow {
+  // TODO: Convert into base class and derive consumer, producer, and hybrid
+  // base classes
+
+public:
+  IceFlow(const std::string &syncPrefix, const std::string &nodePrefix,
+          const std::optional<std::string> &subTopic,
+          const std::optional<std::string> &pubTopic,
+          const std::vector<int> &nTopic, ndn::Face &face,
+          std::optional<int> publishInterval)
+      : m_face(face), m_scheduler(face.getIoContext()),
+        m_nodePrefix(nodePrefix), m_pubTopic(pubTopic), m_subTopic(subTopic),
+        m_topicPartitions(nTopic), m_publishInterval(publishInterval) {
```

Comment:
`nTopic` seems counter-intuitive if the passed argument is not an integer specifying the number of partitions.

```suggestion
          const std::vector<int> &topicPartitions, ndn::Face &face,
          std::optional<int> publishInterval)
      : m_face(face), m_scheduler(face.getIoContext()),
        m_nodePrefix(nodePrefix), m_pubTopic(pubTopic), m_subTopic(subTopic),
        m_topicPartitions(topicPartitions), m_publishInterval(publishInterval) {
```